addons/account/models/account_move.py

```diff
@@ -2212,13 +2212,17 @@ def is_internal_partner(partner):
             # Helper to know if the partner is an internal one.
             return partner.user_ids and all(user.has_group('base.group_user') for user in partner.user_ids)
 
+        extra_domain = False
+        if custom_values.get('company_id'):
+            extra_domain = ['|', ('company_id', '=', custom_values['company_id']), ('company_id', '=', False)]
+
         # Search for partners in copy.
         cc_mail_addresses = email_split(msg_dict.get('cc', ''))
-        followers = [partner for partner in self._mail_find_partner_from_emails(cc_mail_addresses) if partner]
+        followers = [partner for partner in self._mail_find_partner_from_emails(cc_mail_addresses, extra_domain) if partner]
```

Comment:
You forgot to name the `extra_domain` argument, instead they're passed positionally as `records`, so I don't think this PR did anything.

Was causing issues after updating a [forward port](https://github.com/odoo/odoo/pull/159108#event-12234988623) to use `_mail_get_company_field` on `records`, so created a backport PR for the fix https://github.com/odoo/odoo/pull/159265